test/extended/etcd/invariants.go

```diff
@@ -49,3 +72,59 @@ var _ = g.Describe("[sig-etcd] etcd", func() {
 		}
 	})
 })
+
+func getCurrentNetworkTopology(oc *exutil.CLI) string {
+	configClient, err := configv1client.NewForConfig(oc.AdminConfig())
+	o.Expect(err).ToNot(o.HaveOccurred())
+	nework, err := configClient.ConfigV1().Networks().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	o.Expect(err).ToNot(o.HaveOccurred())
+	ipFamily, err := getPreferredIPFamily(nework)
+	o.Expect(err).ToNot(o.HaveOccurred())
+	return ipFamily
+}
+
+// GetPreferredIPFamily checks network status for service CIDR to conclude IP family. If status is not yet populated fallback to spec.
+func getPreferredIPFamily(network *configv1.Network) (string, error) {
```

Comment:
took it from https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/dnshelpers/util.go#L76